llvm/lib/Target/SPIRV/SPIRVInstructionSelector.cpp

```diff
@@ -802,8 +784,12 @@ bool SPIRVInstructionSelector::selectExtInst(Register ResVReg,
                      .addImm(static_cast<uint32_t>(Set))
                      .addImm(Opcode);
       const unsigned NumOps = I.getNumOperands();
-      for (unsigned i = 1; i < NumOps; ++i)
-        MIB.add(I.getOperand(i));
+      unsigned Index = 1;
+      if (I.getOperand(Index).getType() ==
```

Comment:
Might be worth an assert that `NumOps > 2` so folks get a better/quicker error if they accidentally misuse this

Reply:
No, I think `NumOps > 1` is right - we're just making sure that the `I.getOperand(Index)` here (where `Index == 1`) will crash in a helpful way. This makes the assumption here more explicit.

Reply:
Sorry misunderstood. I thought you wanted the assert in the if block. this makes more sense.